lib/sles4sap.pm

```diff
@@ -350,23 +350,27 @@ sub copy_media {
     # First copy media
     my $mnt_path = '/mnt';
     assert_script_run "mkdir $target";
-    assert_script_run "mount -t $proto -o ro $path $mnt_path";
-    type_string "cd $mnt_path\n";
-    type_string "cd " . get_var('ARCH') . "\n";    # Change to ARCH specific subdir if exists
-    assert_script_run "cp -ax . $target/", $nettout;
+    assert_script_run "mount -t $proto -o ro $path/" . get_var('ARCH') . " $mnt_path";
+    assert_script_run "cp -ax $mnt_path/. $target/", $nettout;
```

Comment:
I went ahead and approved the PR, but I wanted to comment on this:

I think with this change we're losing some flexibility that was present in the old code, and I'm not sure the advantages of the change outweigh this. Basically, previously we could pass in **HANA** or **NW** a path such as `/path/to/installation/master` as well as `/path/to/installation/master/arch`. With both of those, code would still work, as the `type_string "cd " . get_var('ARCH') . "\n";` command would fail silently if there is not **ARCH** subdir in the share, but allows the test to continue.

With the new code, if we pass a share that has no **ARCH** subdir, we'll get a failure in the test, even if such a share points to a valid HANA or NW installation master.

Advantages I see in the new code are 2:
1. We remove a possible `cd` error in the test results, which again, this was mostly silent and didn't impact the outcome of the test.
2. We remove an interaction with SUT while copying.

But, we lose the flexibility of passing the test a share that it's not formatted with an **ARCH** subdir even if it's a valid share (for example, the shares in kvmmaster). I'm not sure if this is a good idea or not, but will not oppose the change.

Reply:
I agree with you, there are pros and cons with both solutions. As Loic did the test like this, my point of view is to merge it as is. Loic! It's your choice now :smile:

Reply:
Ok, so I think it's better to keep the old behavior in that case. I just changed a little the logic to avoid unneeded `cd` command.
Let me know if it's OK for you.

Reply:
I like it. It preserves the logic, avoids doing a `cd` to determine the **ARCH** subdir exists so there's no silent error in the serial terminal and keeps the same number of interactions with SUT, but replacing a `type_string` with a `script_run` which is better for error handling.

Only change I'd propose would be with the `get_var('ARCH')` in L352: either to switch it to `get_required_var('ARCH')` or to give it a default with `get_var('ARCH', '')`.

Thanks for this change!

Reply:
Last modification done! I will now restart the verification runs.